pkg/reconciler/certificate/resources/cert_manager_certificate.go

```diff
@@ -46,94 +43,72 @@ const (
 	}
 
-	// https://github.com/knative-sandbox/net-certmanager/issues/214
-	// Only use the domain template if the commonName is too big.
+	// Only attempt to do something special if the entry from DNSNames[0] is too big.
 	// This is to make the upgrade path easier and reduce churn on certificates.
 	// The Route controller adds spec.domain to existing KCerts
 	// The KCert controller requests new certs with same domain names, but a different CN if spec.domain is set and the other domain name would be too long
 	// cert-manager Certificates are updated only if the existing domain name kept them from being issued.
 	if len(commonName) > longest {
-		if knCert.Spec.Domain != "" && knCert.Spec.Domain != commonName {
-			//Split out the domain, and create a hash of the remaining part
-			domainSuffix := "." + knCert.Spec.Domain
-			prefix := strings.TrimSuffix(commonName, domainSuffix)
-			if len(prefix) > base36Len {
-				attemptedToShorten = true
-
-				parsedUUID, err := uuid.Parse(string(knCert.UID))
-				if err != nil {
+		//if we have a domain field, we can attempt to shorten, or check if we are dealing with a domainMapping
+		if knCert.Spec.Domain != "" {
+			// if the domain and commonName pulled from DNSNames are the same, we are dealing with a domainmapping
+			if knCert.Spec.Domain == commonName {
+				if len(commonName) > longest {
 					return nil, &apis.Condition{
 						Type:   CreateCertManagerCertificateCondition,
 						Status: corev1.ConditionFalse,
-						Reason: "Failed To Parse UID",
+						Reason: "DomainMapping Name Too Long",
 						Message: fmt.Sprintf(
-							"error creating cert-manager certificate: failed to parse UID (%s) on KCert (%s): %s",
-							knCert.UID,
-							knCert.Name,
-							err,
+							"error creating cert-manager certificate: DomainMapping name (%s) longer than 63 characters",
+							commonName,
 						),
 					}
 				}
-				parsedUUIDbytes := [16]byte(parsedUUID)
-				prefix = strings.ToLower(base36.EncodeBytes(parsedUUIDbytes[:]))
-			}
-			commonName = prefix + domainSuffix
-
-			//If the new name is still too long, then error
-			if len(commonName) > longest {
-				if attemptedToShorten {
+			} else {
+				// we have a domain field and are not a domainMapping
+				// if the domain is too long, even if we shorten, it will still be too big. We should error in that case
+				if len(knCert.Spec.Domain) > (longest - len(Prefix)) {
 					return nil, &apis.Condition{
 						Type:   CreateCertManagerCertificateCondition,
 						Status: corev1.ConditionFalse,
-						Reason: "CommonName Too Long After Shortening",
+						Reason: "Domain Too Long",
 						Message: fmt.Sprintf(
-							"error creating cert-manager certificate: cannot create valid length CommonName: (%s) still longer than 63 characters after shortening",
-							commonName,
+							"error creating cert-manager certificate: Domain (%s)(length: %v) too long, prepending short prefix of (%s)(length: %v) will be longer than 64 bytes",
```

Comment:
Can we use `CommonName` instead of `Domain` - so when people google our error they'll hopefully see it's not specific to Knative

Reply:
ah good point, yeah I can change that